### PR TITLE
Adprime: added new bid params

### PR DESCRIPTION
**Pull request by Adprime**
Added new params: keywords and audiences
doc - https://docs.prebid.org/dev-docs/pbs-bidders.html#adprime

### CODE REVIEW

**Comment by Adprime** [comment]
@VeronikaSolovei9, @hhhjort hi, please pay attention to this pull request, thanks =)

**Comment by hhhjort** [comment]
It is looking good. Perhaps a rebase to the current head will clean up the PR?

**Review by hhhjort — DISMISSED** [review]
LGTM

**Comment by Adprime** [comment]
@guscarreon Thanks. We fixed all problems

File: adapters/adprime/adprime.go [hunk 1]
```diff
@@ -71,19 +71,20 @@ func (a *AdprimeAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *ada
...
 		// keywords
-		if len(adprimeExt.Keywords) > 0 {
-			if reqCopy.Site == nil {
-				reqCopy.Site = &openrtb2.Site{}
-			}
-			reqCopy.Site.Keywords = strings.Join(adprimeExt.Keywords, ",")
+		if reqCopy.Site != nil && len(adprimeExt.Keywords) > 0 {
+			siteCopy := *reqCopy.Site
+			siteCopy.Keywords = strings.Join(adprimeExt.Keywords, ",")
+			reqCopy.Site = &siteCopy
```
Comment (guscarreon) [inline adapters/adprime/adprime.go:77]: This is a slight change in logic when compared to f4166f4f, right? The f4166f4f version would copy the keywords found in `adprimeExt.Keywords` even if `reqCopy.Site` is `nil`. Are we sure we don't want to keep that logic and do something similar to the logic inside the `if len(adprimeExt.Audiences) > 0` in line 81?

**Review by guscarreon — COMMENTED** [review]
@Adprime thank you for addressing our feedback. I left one last question but your adapter looks pretty solid.

File: adapters/adprime/adprime.go [hunk 2]
```diff
@@ -129,6 +169,9 @@ func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType,
 			if imp.Banner == nil && imp.Video != nil {
 				mediaType = openrtb_ext.BidTypeVideo
 			}
+			if imp.Banner == nil && imp.Video == nil && imp.Native != nil {
+				mediaType = openrtb_ext.BidTypeNative
+			}
```
[the remark this answers is not in the document]
Comment (hhhjort) [reply adapters/adprime/adprime.go:177]: Its not supposed to be possible that all supported types are missing. But extra safety does not hurt.

**Review by hhhjort — DISMISSED** [review]

File: adapters/adprime/adprime.go (hunk 1 above)
Comment (Adprime) [reply adapters/adprime/adprime.go:77]: thanks for paying attention to this, we fixed the code as needed

Comment (guscarreon) [reply adapters/adprime/adprime.go:77]: Do we want to check for `reqCopy.Site != nil` in both lines 74 and 81? Or did you mean to check for `reqCopy.User != nil` in line 81 instead?

On a second thought maybe my comment on my last review was misunderstood. I intention was to ask whether or not you'd want to keep initializing `reqCopy.Site` whenever there were elements inside `adprimeExt.Keywords` even if it was `nil`, just like before. I meant something like:
```
73         // keywords
74 -       if reqCopy.Site != nil && len(adprimeExt.Keywords) > 0 {
   +       if len(adprimeExt.Keywords) > 0 {
   +           if reqCopy.Site == nil {
   +               reqCopy.Site = &openrtb2.Site{}
   +           }
75             siteCopy := *reqCopy.Site
76             siteCopy.Keywords = strings.Join(adprimeExt.Keywords, ",")
77             reqCopy.Site = &siteCopy
78         }
79
80         // audiences
81 -       if reqCopy.Site != nil && len(adprimeExt.Audiences) > 0 {
   +       if len(adprimeExt.Audiences) > 0 {
82             if reqCopy.User == nil {
83                 reqCopy.User = &openrtb2.User{}
84             }
85             userCopy := *reqCopy.User
86             userCopy.CustomData = strings.Join(adprimeExt.Audiences, ",")
87             reqCopy.User = &userCopy
88         }
adapters/adprime/adprime.go
```
Sorry to be so inquisitive about this part of the code @Adprime

**Review by guscarreon — COMMENTED** [review]
Hi @Adprime left one last comment in regards to `Site != nil`. Let me know your take
https://github.com/prebid/prebid-server/pull/1963#discussion_r700332435

File: adapters/adprime/adprime.go (hunk 1 above)
Comment (Adprime) [reply adapters/adprime/adprime.go:77]: Hi, `keywords` and `audiences` must be recorded in the `Site` and `User` objects, respectively, only on the condition that the "Site" object is in the bid request. If the `Site` object is present, but the `User` object is not present, then we will create it.

Comment (guscarreon) [reply adapters/adprime/adprime.go:77]: I understand. Thank you for clarifying.

**Review by guscarreon — APPROVED** [review]
Looks good to me. @Adprime  thank you for addressing our feedback.

**Review by hhhjort — APPROVED** [review]
LGTM

**Pull request merged**